### ISSUE TITLE
jsxLoader: Less than operator assumed to be an open element

### ISSUE THREAD

**Issue by nifgraup** [dataformsjs/dataformsjs#20]
```
<html>
  <head>
    <script src="https://cdn.jsdelivr.net/npm/dataformsjs@5.10.4/js/react/jsxLoader.min.js"></script>
  </head>
<body>
    <script type="text/babel" data-type="module">
      const n = 2;
      1<n;
    </script>
  </body>
</html>
```
```
jsxLoader.min.js:5 Error: The number of opening elements (for example: "<div>") does not match the number closing elements ("</div>").
    at Object.parser (jsxLoader.min.js:5)
    at Object.compile (jsxLoader.min.js:5)
    at r (jsxLoader.min.js:5)
    at jsxLoader.min.js:5
    at new Promise (<anonymous>)
    at Object.loadScript (jsxLoader.min.js:5)
    at Object.setup (jsxLoader.min.js:5)
    at HTMLDocument.<anonymous> (jsxLoader.min.js:5)
r @ jsxLoader.min.js:5
(anonymous) @ jsxLoader.min.js:5
loadScript @ jsxLoader.min.js:5
setup @ jsxLoader.min.js:5
(anonymous) @ jsxLoader.min.js:5
```
Perhaps it's best to skip this test altogether?

**Comment by ConradSollitt** [comment]
Hi Björgvin,

Thanks for trying out the JSX Loader and opening this issue with a simple example.

I will later look into improving how expressions are handled. Most likely I will work on this next week but it may take longer to fix.

```js
// Adding a space before the variable would allow code to be compiled correctly
1 < n;
// Additionally if a character other than a letter
// is after [<] then code compiles correctly as well.
n<1;
```

The specific error message is to help with conditions like there where there are multiple opening elements but zero or fewer closing elements.
```jsx
  // Error Example
  function App() {
    return <div>Hello World<div>
  }
  // Valid because of closing </div>
  function App() {
    return <div>Hello World</div>
  }
```

The actual issue is caused by the compiler checking for a letter after a `<` character and if found it assumes the start of an element is found. To fix this I'll have to add more logic to determine if position in code may be part of an expression. Here is an example from the compiler where it is checking the character and next character.

https://github.com/dataformsjs/dataformsjs/blob/master/js/react/jsxLoader.js#L787
```
  case '<':
    if (/[a-zA-Z>]/.test(input[c + 1])) {
      return c; // Start of Element found
    }
```

**Comment by nifgraup** [comment]
You're welcome, it looks like JSX Loader has a lot of potential. I guess an approach would be to consider any < as potential start of element, enter tokenizeElement and then backtrack or invalidate the token when mismatch happens (instead of throwing an error).

**Comment by ConradSollitt** [comment]
Fixed with the latest release `5.10.6`.
